tests/platforms/linux/appimage/test_mixin.py

```diff
@@ -3,143 +3,243 @@
+    command = LinuxAppImageCreateCommand(
+        logger=Log(),
+        console=Console(),
+        base_path=tmp_path / "base_path",
+        data_path=tmp_path / "briefcase",
+    )
 
     image_tag = command.docker_image_tag(first_app_config)
 
     assert image_tag == f"briefcase/com.example.first-app:py3.{sys.version_info.minor}"
 
 
 def test_docker_image_tag_uppercase_name(uppercase_app_config, tmp_path):
-    command = LinuxAppImageCreateCommand(base_path=tmp_path)
+    command = LinuxAppImageCreateCommand(
+        logger=Log(),
+        console=Console(),
+        base_path=tmp_path / "base_path",
+        data_path=tmp_path / "briefcase",
+    )
 
     image_tag = command.docker_image_tag(uppercase_app_config)
 
     assert image_tag == f"briefcase/com.example.first-app:py3.{sys.version_info.minor}"
 
 
-def test_dockerize(first_app_config, tmp_path):
-    command = LinuxAppImageCreateCommand(base_path=tmp_path)
-    command.Docker = Docker
-    command.use_docker = True
-
-    # Before dockerization, subprocess is native
-    assert type(command.subprocess) == Subprocess
-
-    with command.dockerize(first_app_config):
-        # During dockerization, subprocess is a container
-        assert type(command.subprocess) == Docker
-
-    # After dockerization, subprocess is native
-    assert type(command.subprocess) == Subprocess
-
-
-def test_dockerize_nodocker(first_app_config, tmp_path):
-    """If docker is not in use, dockerize() is a no-op."""
-    command = LinuxAppImageCreateCommand(base_path=tmp_path)
-    command.Docker = Docker
-    command.use_docker = False
-
-    # Before dockerization, subprocess is native
-    assert type(command.subprocess) == Subprocess
-
-    with command.dockerize(first_app_config):
-        # During dockerization, subprocess is *still* native
-        assert type(command.subprocess) == Subprocess
-
-    # After dockerization, subprocess is native
-    assert type(command.subprocess) == Subprocess
-
-
-def test_verify_linux_no_docker(tmp_path):
-    """If Docker is disabled on Linux, the Docker alias is not set."""
-    command = LinuxAppImageCreateCommand(base_path=tmp_path)
-    command.host_os = "Linux"
+def test_verify_linux_no_docker(tmp_path, first_app_config):
+    """If Docker is disabled on Linux, the app_context is Subprocess."""
+    command = LinuxAppImageCreateCommand(
+        logger=Log(),
+        console=Console(),
+        base_path=tmp_path / "base_path",
+        data_path=tmp_path / "briefcase",
+    )
+    command.tools.host_os = "Linux"
     command.use_docker = False
 
     # Verify the tools
     command.verify_tools()
+    command.verify_app_tools(app=first_app_config)
 
-    # No error, but no Docker wrapper either.
-    assert command.Docker is None
+    # No error and Subprocess is used.
+    assert isinstance(command.tools[first_app_config].app_context, Subprocess)
+    # Docker is not verified.
+    assert getattr(command.tools, "docker", "MISSING") == "MISSING"
```

Comment:
`assert hasattr()` would be another option here.

Reply:
Missed replacing this one. Thanks.